File: C++/Curso Estruct. Abs. De Datos y Algoritmos/Laboratorios/Lab3GeometriaPython/sourcecode/Imports/Vertice.py
#@file Vertice.py

#@brief Clase que crea vertices en el espacio 2D
class Vertice:
    #@brief constructor de vertice
    def __init__(self):
        self.x=0 #Valor para la coordenada x en la que se encuentra el vertice
        self.y=0 #Valor para la coordenada y en la que se encuentra el vertice
        self.identificador = 0
    
    #@brief Sobrecarca de operador  ">>" calcula la distancia entre dos vertices
    #@param other es el otro vertice
    def __rshift__(self, other): 
        distancia = 0
        xcuadrado = (self.x - other.x)
        ycuadrado = (self.y - other.y)
        xcuadrado = (xcuadrado)**2
        ycuadrado = (ycuadrado)**2
        distancia = (xcuadrado + ycuadrado)**(1/2)
        return distancia

    #@brief Sobrecarca de operador  "~" representacion de texto de un vertice 
    #@return Devuelve la representacion en texto de un vertice
    def __invert__(self):
        respuesta = "Coordenada en X es:" + str(self.x) + " y mi Coordenada en Y es:" + str(self.y)
        return respuesta

File: C++/Curso Estruct. Abs. De Datos y Algoritmos/Laboratorios/lab2 Geometria/sourcecode/Vertice.cpp
/**
 *@file Vertice.cpp/
 */
#include "../include/Includes.hpp"
using namespace std;

/**
 * @brief constructor por defecto
 */
Vertice::Vertice(){
}
/**
 * @brief desconstructor por defecto
 */
Vertice::~Vertice(){
}
/**
 * @brief constructor por copia
 * @param v es un objeto
 */
Vertice::Vertice(const Vertice& v){
    this->x=v.x;
    this->y=v.y;
    this->identificador = v.identificador;
}
/**
 * @brief sobrecarga operador ~ para convertir a texto
 */
string Vertice::operator~(){
    string respuesta = "";
    string strX = to_string(x);
    string strY = to_string(y);
    string identi = to_string(identificador);
    respuesta = ("punto: " + identi + " Coordenada X "+ strX +" coordenada Y "+ strY);
    return respuesta;
}
/**
 * @brief sobrecarga operador ~ para convertir a texto
 */
double Vertice::operator>>(const Vertice& rhs){
    float distancia = 0;
    float xcuadrado = (this->x - rhs.x);
    float ycuadrado = (this->y - rhs.y);
    xcuadrado = pow(xcuadrado,2);
    ycuadrado = pow(ycuadrado,2);
    distancia = sqrt (xcuadrado + ycuadrado);
    return distancia;
}
